Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs

```diff
@@ -160,6 +184,16 @@ public ActionResult Edit(ContestAdministrationViewModel model)
 
                 contest = model.GetEntityModel(contest);
 
+                if (contest.IsOnline() &&
+                    contest.IsActive &&
+                    (contest.Duration != model.Duration ||
+                        contest.NumberOfProblemGroups != model.NumberOfProblemGroups ||
+                        (int)contest.Type != model.Type))
+                {
+                    this.TempData.AddDangerMessage(Resource.Active_contest_cannot_edit_duration_type_problem_groups);
+                    this.RedirectToAction<ContestsController>(c => c.Index());
+                }
+
```

Comment:
Add additional validation for the `Duration` field as it can easily overflow.

Reply:
Done